packages/stylis/src/index.js

```diff
@@ -1,2 +1,1215 @@
+              return serialize(
+                [
+                  copy(
+                    replace(value, /(plac.+)/, WEBKIT + 'input-$1'),
+                    element,
+                    ''
+                  ),
+                  copy(replace(value, /(plac.+)/, MOZ + '$1'), element, ''),
+                  copy(
+                    replace(value, /:(plac.+)/, MS + 'input-$1'),
+                    element,
+                    ''
+                  )
+                ],
+                callback
+              )
+          }
+
+          return ''
+        })
+  }
+}
+
+/**
+ * @param {object} element
+ */
+function namespace(element) {
+  switch (element.type) {
+    case RULESET:
+      element.props = element.props.map(function(value) {
+        return combine(tokenize(value), function(value, index, children) {
+          switch (charat(value, 0)) {
+            // \f
+            case 12:
+              return substr(value, 1, strlen(value))
+            // \0 ( + > ~
+            case 0:
+            case 40:
+            case 43:
+            case 62:
+            case 126:
+              return value
+            // :
+            case 58:
+              if (children[index + 1] === 'global')
+                (children[index + 1] = ''),
+                  (children[index + 2] =
+                    '\f' + substr(children[index + 2], (index = 1), -1))
+            // \s
+            case 32:
+              return index === 1 ? '' : value
+            default:
+              switch (index) {
+                case 0:
+                  element = value
+                  return sizeof(children) > 1 ? '' : value
+                case (index = sizeof(children) - 1):
+                case 2:
+                  return index === 2
+                    ? value + element + element
+                    : value + element
+                default:
+                  return value
+              }
+          }
+        })
+      })
+  }
+}
+
+/**
+ * @param {string[]} tokens
+ * @param {number} current
+ */
+function move(tokens, current) {
+  while (++current < tokens.length) {
+    if (charat(tokens[current], 0) === 44) break
+  }
+  return ++current
+}
+
+/**
+ * @param {object} element
+ */
+function compat(element) {
+  switch (element.type) {
+    case RULESET:
+      var current = 0
+      var next = 0
+      var props = element.props
+      var value = element.value
+      var length = props.length
+      var prop
+
+      if (length === 1 && charat(value, 0) !== 58) {
+        return
+      }
+
+      var tokens = tokenize(value)
+
+      for (var i = 0; i < length; i++) {
+        ;(prop = props[i]), (next = move(tokens, (current = next)))
+
+        if (charat(tokens[current], 0) !== 58) {
+          continue
+        }
+
+        var pattern = ''
+        while (current < next - 1) {
+          pattern += tokens[current++]
+        }
+        props[i] = replace(prop, ' ' + pattern, pattern)
+      }
+  }
+}
+
+export {
```

Comment:
Do we really want to export all of this?

Reply:
Nah, it's just a copy paste from the stylis itself right now as v4 is not yet published to npm. I think it should be fine to just use the stylis itself and make `@emotion/stylis` obsolete.